### ISSUE TITLE
pio train didn't work on macOS. It fails without any message

### ISSUE THREAD

**Issue by ishantanu** [PredictionIO/template-scala-parallel-universal-recommendation#52]
This is what I got when I ran pio train.

/usr/local/opt/apache-spark is probably an Apache Spark development tree. Please make sure you are using at least 1.3.0.
[INFO] [Console$] Using existing engine manifest JSON at /usr/local/Cellar/universal-recommender/manifest.json
[INFO] [Runner$] Submission command: /usr/local/opt/apache-spark/bin/spark-submit --class io.prediction.workflow.CreateWorkflow --jars file:/usr/local/Cellar/universal-recommender/target/scala-2.10/template-scala-parallel-universal-recommendation-assembly-0.2.3-deps.jar,file:/usr/local/Cellar/universal-recommender/target/scala-2.10/template-scala-parallel-universal-recommendation_2.10-0.2.3.jar --files file:/usr/local/Cellar/predictionio/0.9.6/libexec/conf/log4j.properties --driver-class-path /usr/local/Cellar/predictionio/0.9.6/libexec/conf:/usr/local/opt/elasticsearch/config:/usr/local/Cellar/predictionio/0.9.6/libexec/lib/postgresql-9.4-1204.jdbc41.jar:/usr/local/Cellar/predictionio/0.9.6/libexec/lib/mysql-connector-java-5.1.37.jar file:/usr/local/Cellar/predictionio/0.9.6/libexec/lib/pio-assembly-0.9.6.jar --engine-id ZayKJWtwNT8xBsBN8d0ggoQrsvKfKc0v --engine-version 1dcba07d591527a2cd6b76708ba398ff5c310b78 --engine-variant file:/usr/local/Cellar/universal-recommender/engine.json --verbosity 0 --json-extractor Both --env PIO_ENV_LOADED=1,PIO_ACCESS_KEY=rK2qpEOQPGQJNKqblTk1afdq4nG7vujx34nv3r4AhoRgcJ0u5zEBLJf2XAkrqW5O,PIO_STORAGE_REPOSITORIES_METADATA_NAME=pio_meta,PIO_FS_BASEDIR=/Users/shantanudeshpande/.pio_store,PIO_HOME=/usr/local/Cellar/predictionio/0.9.6/libexec,PIO_FS_ENGINESDIR=/Users/shantanudeshpande/.pio_store/engines,PIO_STORAGE_SOURCES_PGSQL_URL=jdbc:postgresql://localhost/pio,PIO_STORAGE_REPOSITORIES_METADATA_SOURCE=PGSQL,PIO_STORAGE_REPOSITORIES_MODELDATA_SOURCE=PGSQL,PIO_STORAGE_REPOSITORIES_EVENTDATA_NAME=pio_event,PIO_STORAGE_SOURCES_PGSQL_PASSWORD=pio,PIO_STORAGE_SOURCES_PGSQL_TYPE=jdbc,PIO_FS_TMPDIR=/Users/shantanudeshpande/.pio_store/tmp,PIO_STORAGE_SOURCES_PGSQL_USERNAME=pio,PIO_STORAGE_REPOSITORIES_MODELDATA_NAME=pio_model,PIO_STORAGE_REPOSITORIES_EVENTDATA_SOURCE=PGSQL,PIO_CONF_DIR=/usr/local/Cellar/predictionio/0.9.6/libexec/conf

**Comment by nullcrimson** [comment]
The same issue for me. Are there any workarounds?

**Comment by ishantanu** [comment]
Got it working when I manually set up PredictionIO from scratch and start over again. Its working now.

**Comment by nullcrimson** [comment]
@iamShantanu101 So you downloaded the source code without using brew and managed to get that up and running? Do I have to install all the dependencies manually as well or should I use brew?

**Comment by ishantanu** [comment]
@RomanKapitonov You can try building PredictionIO from source. Thats what worked for me. There might be better solutions but yeah this is what will definitely work. Regarding dependencies, which ones are you talking about? Elasticsearch,postgres,etc?

**Comment by nullcrimson** [comment]
@iamShantanu101 Yes, what about elasticsearch, spark and hbase/mysql/pg packages? BTW are you running 0.10.0 or an older version?

**Comment by pferrel** [comment]
This looks like you have seriously broken configuration of your system. 

BTW PredictionIO is not hosted here, and neither is the Universal recommender. PIO is now Apache-PredictionIO and the UR is several revs beyond this repo.

For Apache PIO v0.10.0-incubating use this template: https://github.com/actionml/universal-recommender. You will need to download and build Mahout from this repo: https://github.com/apache/mahout.git checkout this commit: 00a2883ec69b0807a5486c61dfcc7ef27f35ddc6

Build Apache PIO, then Mahout, then the UR. This will al least put you on a version we can support. Then look for support, for the UR here: https://groups.google.com/forum/#!forum/actionml-user

Sorry for all the changes but Salesforce bought the company that owned PIO and made it an Apache project, this meant changes in every class inside PIO due to being named for Apache.org. That in turn means changes to every template.

**Comment by pferrel** [comment]
the UR pointed to in the above answer will only work with Apache PIO v0.10.0

Not sure if the installation script works for the dependencies. I have a Mac and would not use brew for Spark, Hadoop, HBase, and Elasticsearch. You need to have the ability to move to the right versions and starting with brew sort of locks you into their releases. I use brew for Python and other things but the above deps, no.

**Comment by ishantanu** [comment]
Exactly thats what I was going to say for dependencies. I too opted not to use brew for this purpose. Moreover if you installed predictionio using brew, then it might be the case that PIO is v0.9.6 rather than v0.10.0

**Comment by nullcrimson** [comment]
@pferrel @iamShantanu101 Thank you guys for clarification. I appreciate that.

For anyone interested or facing the same issue: I've been able to setup everything using the [following documentation](http://actionml.com/docs/pio_quickstart) (quick install if to be exact). Manual installation seems to be pretty straighforward if it is your case.

Note that you will need maven installed locally to be able to build the required version of Mahout. The documentation above describes how to do that. After all I have predictionio v0.9.7-aml  up and running with postgresql backend
